server/src/main/java/org/cloudfoundry/identity/uaa/impl/config/AwsSecretsLoader.java

```diff
@@ -0,0 +1,86 @@
+package org.cloudfoundry.identity.uaa.impl.config;
+
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueResult;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.awspring.cloud.core.SpringCloudClientConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class AwsSecretsLoader {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsSecretsLoader.class);
+
+    public void loadSecrets(ConfigurableEnvironment environment) {
+        String awsSecretNames = environment.getProperty("AWS_SECRET_MANAGER_NAMES");
+        if (StringUtils.hasText(awsSecretNames)) {
+            String[] secretNames = awsSecretNames.split(";");
+            AWSSecretsManager secretManagerClient = awsSecretsManager();
+            Arrays.stream(secretNames).forEach(secretName -> {
+                try {
+                    LOGGER.info("Loading secrets from AWS Secret Manager for Secret Name {}", secretName);
+
+                    GetSecretValueResult results =
+                            secretManagerClient.getSecretValue(new GetSecretValueRequest().withSecretId(secretName));
+                    JsonNode jsonNode = mapper.readTree(results.getSecretString());
+                    Map<String, Object> map = mapper.convertValue(jsonNode, Map.class);
+                    Map<String, Object> finalMap = new LinkedHashMap<>();
+                    map.forEach((key, value) -> {
+                        if (value instanceof Map) {
+                            Map<String, Object> valFromEnv = environment.getProperty(key, Map.class);
+                            if (valFromEnv != null) {
+                                deepMergeMaps((Map<String, Object>) value, valFromEnv);
```

Comment:
Is it possible better add everything to finalMap and write it to environment at once? instead of directly working on the spring environment object.